src/main/java/org/betonquest/betonquest/item/typehandler/SpigotHeadHandler.java

```diff
@@ -0,0 +1,75 @@
+package org.betonquest.betonquest.item.typehandler;
+
+import lombok.CustomLog;
+import org.betonquest.betonquest.api.profiles.Profile;
+import org.bukkit.Bukkit;
+import org.bukkit.inventory.meta.SkullMeta;
+import org.bukkit.profile.PlayerProfile;
+
+import java.net.URL;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@CustomLog
+@SuppressWarnings({"PMD.CommentRequired", "deprecation"})
+public class SpigotHeadHandler extends HeadHandler {
+    /**
+     * Parse the metadata of a SkullMeta instance that needs to be persisted so that it can be correctly reconstituted.
+     * @param skullMeta The SkullMeta to parse.
+     * @return A Map of the properties parsed from the SkullMeta.
+     */
+    public static Map<String, String> parseSkullMeta(final SkullMeta skullMeta) {
+        final Map<String, String> parsedValues = new HashMap<>();
+        if (skullMeta.hasOwner()) {
+            parsedValues.put(META_OWNER, skullMeta.getOwner());
+        }
+        return parsedValues;
+    }
+
+    @Override
+    public void populate(final SkullMeta skullMeta, final Profile profile) {
+        final String owner = getOwner(profile);
+        final UUID playerId = getPlayerId();
+        final String texture = getTexture();
+
+        if (owner != null) {
+            skullMeta.setOwner(owner);
+        }
+        if (playerId != null && texture != null) {
+            try {
+                final URL url = new URL(new String(Base64.getDecoder().decode(texture)));
+
+                final PlayerProfile playerProfile = Bukkit.getServer().createPlayerProfile(playerId);
+                playerProfile.getTextures().setSkin(url);
+                skullMeta.setOwnerProfile(playerProfile);
+            } catch (final Exception e) {
+                LOG.error("Error determining texture URL", e);
+            }
+        }
+    }
+
+    @Override
+    public boolean check(final SkullMeta skullMeta) {
+        final String ownerName = skullMeta.getOwner();
+        final PlayerProfile playerProfile = skullMeta.getOwnerProfile();
+        if (playerProfile != null) {
+            final UUID playerUniqueId = playerProfile.getUniqueId();
+            final URL skin = playerProfile.getTextures().getSkin();
+            if (skin != null) {
+                final String texture = toEncodedSkinJson(skin);
+                return checkOwner(ownerName) && checkPlayerId(playerUniqueId) && checkTexture(texture);
+            } else {
+                return checkOwner(ownerName);
+            }
+        } else {
+            return checkOwner(ownerName);
+        }
+    }
+
+    private static String toEncodedSkinJson(final URL skinUrl) {
```

Comment:
this is not really converting to json. I changed my mind during implementation to simplify. I can change this method name.